patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -134,6 +134,20 @@
  		public static string worldName = "";
  		public static int worldID;
  		public static int background;
+@@ -820,7 +_,12 @@
+ 		public static float invAlpha = 1f;
+ 		public static float invDir = 1f;
+ 		[ThreadStatic]
++		private static UnifiedRandom _rand;
+-		public static UnifiedRandom rand;
++		public static UnifiedRandom rand
++		{
++			get => _rand ??= new UnifiedRandom((int)DateTime.Now.Ticks);
++			set => _rand = value;
++		}
```

Comment:
This isn't autoloading. Also why are we changing this? Synchronisation?

Reply:
I guess it can help ensuring the random generator is always there?

Reply:
This is a fix that should've been on 1.4 branch, and I'll recommit it there right now because everything is essentially broken without this.